app/models/case_log.rb

```diff
@@ -118,6 +117,22 @@ def postcode_full
     end
   end
 
+  def postcode_full=(postcode)
+    if postcode
+      super UKPostcode.parse(postcode).to_s
+    else
+      super nil
```

Comment:
passing nil to be library crashes the parser

Reply:
Can this be extracted as a method? We've got the same code in 3 places

Reply:
Does super need to run for the nil case? Can it not just be:

```ruby
  super UKPostcode.parse(postcode) if postcode
```

Reply:
I think we can assign a nil value to some fields, I think that's what the form does when it invalidates some fields.